### ISSUE TITLE
Parser should interpret adjacent parentheses as multiplication

### ISSUE THREAD

**Issue by nicolewhite** [nicolewhite/algebra.js#14]
The parser interprets the following incorrectly:

```js
var exp = algebra.parse("(x + 2)(x + 2)");
exp.toString(); // x + 2
```

Currently, you have to do it like this:

```js
var exp = algebra.parse("(x + 2) * (x + 2)");
exp.toString(); // x^2 + 4x + 4
```

But it should be able to parse the top example too. @timrach can you take a look?

**Comment by kevinb7** [comment]
It should probably handle cases like `5(x+2)` and `(x+2)5`.

**Comment by timrach** [comment]
Is fixed in #17
